Switch validator over to load attestation and block duties separately

ajsutton opened the pull request:
## PR Description
Use the separate APIs to load attestation and block duties rather than the older combined one.

## Fixed Issue(s)
#1918 

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

ajsutton committed:
Switch validator over to load attestation and block duties separately.

ajsutton committed:
Merge branch 'master' into new-duty-loaders

rolfyone commented on validator/client/src/main/java/tech/pegasys/teku/validator/client/AbstractDutyLoader.java:
```diff
@@ -0,0 +1,73 @@
+    this.scheduledDutiesFactory = scheduledDutiesFactory;
+    this.validators = validators;
+    this.validatorIndexProvider = validatorIndexProvider;
+    // TODO: Find a new place to set the local_validator_count metric
```
todo without ticket number..

ajsutton replied:
Restored the metric.

rolfyone reviewed (approved):
LGTM, just a nit.

ajsutton committed:
Restore validator count metric.

ajsutton committed:
Merge branch 'master' of github.com:PegaSysEng/teku into new-duty-loaders

ajsutton committed:
Merge branch 'new-duty-loaders' of github.com:ajsutton/teku into new-duty-loaders

ajsutton committed:
Replace epoch in URL correctly.
Include url in error message when a request fails.
Map attester duty response to internal representation correctly.

ajsutton committed:
Add tests for the duties api client.

ajsutton committed:
Throw exception when a bad request response is received so the RetryingDutyLoader can handle it rather than suppressing the exception and reporting chain data unavailable.

ajsutton committed:
Errorprone.
